src/master-playlist-controller.js

```diff
@@ -289,6 +291,23 @@ export class MasterPlaylistController extends videojs.EventTarget {
     } else {
       this.masterPlaylistLoader_.load();
     }
+    const timeToCanPlayStart = Date.now();
+
+    this.timeToFirstFrame__ = 0;
+    this.appendsToFirstFrame__ = 0;
+
+    this.tech_.one('canplay', () => {
```

Comment:
once we 'canplay' a frame will be showing.

Reply:
Apparently this is close to a "time to first byte" since if we are paused, the video may not get rendered immediately. `playing` or first `timeupdate` is potentially closer to a time to first rendered frame if we are autoplaying/not paused.

Reply:
There's this experimental API that can better represent the frame getting rendered https://wicg.github.io/video-rvfc/

Reply:
switched it to `loadeddata` https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/loadeddata_event
as it specifically says: "The loadeddata event is fired when the frame at the current playback position of the media has finished loading; often the first frame."

Reply:
that seems more like time to first byte rather than time to first frame, the latter implies that the frame has rendered which loadeddata doesn't guarantee it.

Reply:
Yup, basically, this isn't a simple change. We should figure out what we care about and do our best to capture that.

Preload value also plays into account here.

Also, I believe artificial timeupdates we fire only fire on the player and not the tech.

Reply:
I switched this over to using timeupdate and took into account preload.

Reply:
I guess what do you consider `timeToFirstFrame`? Is it the data has been loaded into the video element or the frame has rendered visibly to the user?

Reply:
Since we don't have a good way to determine if the frame has been rendered visibly to the user (except for the experimental API), I think time to first frame should be when the video element reports that it has a `buffered` region containing `currentTime`. That might be a few ms before the frame is visible to the user, but I think it's the best cross browser solution.

Reply:
Makes sense. I would probably want to rename the field then. We should also make sure to document exactly what it means so that we don't get confused in the future and have a place to reference it.
Also, does live have anything special for this beyond being equivalent to preload=none?

Reply:
maybe `timeToLoadedData` and `appendsToLoadedData`? Live shouldn't be any different for VOD for these stats.